### PR TITLE
Password  validation

### PR DESCRIPTION
**Pull request by nickywilliams114**
added a password validation

### CODE REVIEW

File: StokedSportsWebApi/wwwroot/index.js [hunk 1]
```diff
@@ -1,18 +1,44 @@
...
+function login(event) {
+    //store the username
+    //store password
+    //validate username and password
+    //send request to server to get a token
+    //take user to secure page
+
+    let email = document.getElementById("txtEmail").value;
+    let password = document.getElementById("txtPassword").value;
+}
```
Comment (bsdisaster) [inline StokedSportsWebApi/wwwroot/index.js:12]: Link code you should run this code in the login method. 
On line 12 add `If(!validatePassword) { return; }`

File: StokedSportsWebApi/wwwroot/index.js [hunk 2]
```diff
@@ -1,18 +1,44 @@
...
+function validatePassword(password) {
+    var password = /^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{7,15}$/;
```
Comment (bsdisaster) [inline StokedSportsWebApi/wwwroot/index.js:14]: In validatePassword. Return True if password is good and meets criteria. Return False if it is not good.

Comment (bsdisaster) [reply StokedSportsWebApi/wwwroot/index.js:14]: Suggestion on checking password 
let passwordLength = password.length
if(passwordLength < 8) { return false }
\

**Review by AustenCouch — COMMENTED** [review]
on line 16, an alert just saying "false, try again" might not be descriptive enough. Perhaps something like "Invalid password, please enter a valid password"

File: StokedSportsWebApi/wwwroot/index.js [hunk 3]
```diff
@@ -1,18 +1,55 @@
-﻿
-async function sendAuthorizedRequestAsync(apiUrl, methodType, data) {
```
Comment (AustenCouch) [inline StokedSportsWebApi/wwwroot/index.js:2]: why was line 2 removed? Just to make sure this wasn't an accident.

**Review by AustenCouch — COMMENTED** [review]
Looked good to me.

File: StokedSportsWebApi/wwwroot/index.js (hunk 3 above)
Comment (nickywilliams114) [reply StokedSportsWebApi/wwwroot/index.js:2]: Hmm, I''m not sure.  I think it was by accident.

**Pull request merged**